plugins/talk-plugin-local-auth/server/mutators.js

```diff
@@ -35,24 +35,22 @@ async function updateUserEmailAddress(ctx, email, confirmPassword) {
   email = email.toLowerCase().trim();
 
   // Update the Users email address.
-  try {
-    await User.update(
-      {
-        id: user.id,
-        profiles: { $elemMatch: { provider: 'local' } },
-      },
-      {
-        $set: { 'profiles.$.id': email },
-        $unset: { 'profiles.$.metadata.confirmed_at': 1 },
-      }
-    );
-  } catch (err) {
+  await User.update(
+    {
+      id: user.id,
+      profiles: { $elemMatch: { provider: 'local' } },
+    },
+    {
+      $set: { 'profiles.$.id': email },
+      $unset: { 'profiles.$.metadata.confirmed_at': 1 },
+    }
+  ).catch(err => {
```

Comment:
As this is in an async function, it should (and from my tests, currently does) already throw the `ErrEmailTaken` error.

Reply:
You are right. I've tested it on 4.4.0, checked on master and It works fine. Thanks